diff --git a/main/index.js b/main/index.js
--- a/main/index.js
+++ b/main/index.js
@@ -47,6 +47,11 @@ Logger_js_1.Logger.setupLogger();
     (0, handleCrash_js_1.handleCrash)();
     await (0, loadURL_js_1.loadURL)(window);
-    if ([platform_js_1.Platform.WINDOWS, platform_js_1.Platform.LINUX].includes(deviceInfo_js_1.devicePlatform)) {
+    const customTitleBar = process.env.CUSTOM_TITLE_BAR && process.env.CUSTOM_TITLE_BAR > 0;
+    const platforms = [platform_js_1.Platform.WINDOWS];
+    if (customTitleBar) {
+        platforms.push(platform_js_1.Platform.LINUX);
+    }
+    if (platforms.includes(deviceInfo_js_1.devicePlatform)) {
         (0, customTitleBar_js_1.createCustomTitleBar)(window);
     }
     if (config_js_1.config.app.enableAutoUpdate) {


diff --git a/main/lib/createWindow.js b/main/lib/createWindow.js
--- a/main/lib/createWindow.js
+++ b/main/lib/createWindow.js
@@ -38,13 +38,14 @@ const toggleWindowState = (window) => {
     (0, tray_js_1.updateTrayMenu)(window);
 };
 exports.toggleWindowState = toggleWindowState;
+const customTitleBar = process.env.CUSTOM_TITLE_BAR && process.env.CUSTOM_TITLE_BAR > 0;
 const createWindow = async () => {
-    const withFrame = [platform_js_1.Platform.WINDOWS, platform_js_1.Platform.MACOS].includes(deviceInfo_js_1.devicePlatform);
+    const withFrame = !customTitleBar;
     const window = new electron_1.BrowserWindow({
         show: false,
         center: true,
         frame: withFrame,
-        titleBarStyle: 'hidden',
+        ...(customTitleBar ? { titleBarStyle: 'hidden'} : {}),
         trafficLightPosition: {
             x: 16,
             y: 10


diff --git a/main/lib/preload.js b/main/lib/preload.js
--- a/main/lib/preload.js
+++ b/main/lib/preload.js
@@ -10,2 +10,3 @@ 
 const devTools = process.env.DEV_TOOLS && process.env.DEV_TOOLS > 0;
+const customTitleBar = process.env.CUSTOM_TITLE_BAR && process.env.CUSTOM_TITLE_BAR > 0;
 electron_1.contextBridge.exposeInMainWorld('VERSION', String(config_js_1.config.buildInfo.VERSION));
@@ -15,2 +16,3 @@ 
 electron_1.contextBridge.exposeInMainWorld('IS_DEVTOOLS_ENABLED', devTools);
+electron_1.contextBridge.exposeInMainWorld('CUSTOM_TITLE_BAR', customTitleBar);
 electron_1.contextBridge.exposeInMainWorld('desktopEvents', {
