diff --git a/main/lib/preload.js b/main/lib/preload.js
--- a/main/lib/preload.js
+++ b/main/lib/preload.js
@@ -25,6 +25,8 @@ electron_1.contextBridge.exposeInMainWorld('desktopEvents', {
         return electron_1.ipcRenderer.invoke(name, ...args);
     }
 });
+const devTools = process.env.DEV_TOOLS && process.env.DEV_TOOLS > 0;
+electron_1.contextBridge.exposeInMainWorld('IS_DEVTOOLS_ENABLED', devTools);
 window.document.addEventListener('DOMContentLoaded', () => {
     const theme = (0, getInitialTheme_js_1.getInitialTheme)();
     if (hostnamePatterns_js_1.applicationHostnamePattern.test(window.location.hostname)) {
