diff --git a/main/lib/handlers/handleWindowLifecycleEvents.js b/main/lib/handlers/handleWindowLifecycleEvents.js
index 269d1f6..ff744b4 100644
--- a/main/lib/handlers/handleWindowLifecycleEvents.js
+++ b/main/lib/handlers/handleWindowLifecycleEvents.js
@@ -36,6 +36,7 @@ const handleWindowLifecycleEvents = (window) => {
         checkAndUpdateApplicationData(window);
     });
     window.on('close', (event) => {
+        electron_1.app.quit();
         if (!state_js_1.state.willQuit) {
             event.preventDefault();
             if (window.isFullScreen()) {
--
2.42.0

