diff --git a/main/index.js b/main/index.js
--- a/main/index.js
+++ b/main/index.js
@@ -34,2 +34,7 @@  exports.config = {
     (0, systemMenu_js_1.setupSystemMenu)(window);
+    const { session } = require("electron");
+
+    session.defaultSession.webRequest.onBeforeRequest({ urls: ["https://yandex.ru/clck/*", "https://mc.yandex.ru/*"] }, (details, callback) => {
+       callback({ cancel: true });
+    });
     const isPlatformWindows = deviceInfo_js_1.devicePlatform === platform_js_1.Platform.WINDOWS;


diff --git a/package.json b/package.json
--- a/package.json
+++ b/package.json
@@ -41,2 +41,2 @@    "appConfig": {
-        "enableAutoUpdate": true,
-        "enableUpdateByProbability": true,
+        "enableAutoUpdate": false,
+        "enableUpdateByProbability": false,
